docs/modules/ROOT/pages/sending-multipart-formdata.adoc

```diff
@@ -0,0 +1,153 @@
+
+    @FormParam("fileName")
+    @PartType(MediaType.TEXT_PLAIN)
+    public String fileName;
+
+    @FormParam("someObject")
+    @PartType(MediaType.APPLICATION_JSON)
+    public MyComplexObject someObject;
+}
+----
+
+Then in the client, when using RESTEasy Classic, the `org.jboss.resteasy.annotations.providers.multipart.MultipartForm` annotation is added in front of the multipart parameter:
+
+[source,java]
+----
+@Path("/echo")
+@RegisterRestClient(baseUri="http://my.endpoint.com/api/v1", configKey="multipart-requests_yml")
+public interface MultipartService {
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    String sendMultipartData(@MultipartForm MultipartBody data);
+
+}
+----
+
+When using RESTEasy Reactive, the `jakarta.ws.rs.BeanParam` annotation is added in front of the multipart parameter:
+
+[source,java]
+----
+@Path("/echo")
+@RegisterRestClient(baseUri="http://my.endpoint.com/api/v1", configKey="multipart-requests_yml")
+public interface MultipartService {
+
+  @POST
+  @Consumes(MediaType.MULTIPART_FORM_DATA)
+  @Produces(MediaType.TEXT_PLAIN)
+  String sendMultipartData(@jakarta.ws.rs.BeanParam MultipartBody data);
+
+}
+----
+
+See https://quarkus.io/guides/rest-client-multipart[Quarkus - Using the REST Client with Multipart] and
+the https://docs.jboss.org/resteasy/docs/4.7.5.Final/userguide/html_single/index.html[RESTEasy JAX-RS specifications] for more details.
+
+WARNING: `MultipartForm`  is deprecated when using RESTEasy Reactive.
+
+`baseURI` value of `RegisterRestClient` annotation is extracted from the `servers` section of the file, if present. If not, it will be left empty and it is expected you set up the uri to be used in your configuration.
+
+Importantly, if some multipart request bodies contain complex objects (i.e. non-primitives) you need to explicitly tell the Open API generator to create models for these objects by setting
+the `skip-form-model` property corresponding to your spec in the `application.properties` to `false`, e.g.:
+
+[source,properties]
+----
+quarkus.openapi-generator.codegen.spec.my_multipart_requests_yml.skip-form-model=false
+----
+
+See the module https://github.com/quarkiverse/quarkus-openapi-generator/tree/main/integration-tests/multipart-request[multipart-request] for an example of how to use this feature.
+
+In case the default `PartFilename` annotation is not required, its generation can be disabled by setting the `generate-part-filename` property (globally or corresponding to your spec) in the `application.properties` to `false`, e.g.:
+
+[source,properties]
+----
+quarkus.openapi-generator.codegen.spec.my_multipart_requests_yml.generate-part-filename=false
+----
+
+By default, the `PartFilename`'s value representing the filename is prefixed by the field name. This can be changed by setting the `use-field-name-in-part-filename` property (globally or corresponding to your spec) in the `application.properties` to `false`, e.g.:
```

Comment:
```suggestion
By default, the `PartFilename` value representing the filename is prefixed by the field name. This can be changed by setting the `use-field-name-in-part-filename` property (globally or corresponding to your spec) in the `application.properties` to `false`, e.g.:
```
I think this quote is breaking the format.